packages/dataviews/src/dataviews-layouts/hooks/use-clickable-item-props.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * External dependencies
+ */
+import clsx from 'clsx';
+
+export const useClickableItemProps = < Item >(
+	item: Item,
+	isItemClickable: ( item: Item ) => boolean,
+	onClickItem: ( item: Item ) => void,
+	className: string
+) => {
+	const isClickable = isItemClickable( item );
+
+	return {
+		className: clsx( className, {
+			[ className + '--clickable' ]: isClickable,
+		} ),
+		role: isItemClickable( item ) ? 'button' : undefined,
+		onClick: () => {
+			if ( isClickable ) {
+				onClickItem( item );
+			}
+		},
+		tabIndex: isClickable ? 0 : undefined,
+		onKeyDown: ( event: React.KeyboardEvent ) => {
```

Comment:
Should make this `undefined` if isClickable is false. Same for onClick prop

Reply:
There is an if inside both functions to ensure that the callback is invoked only when the item is `clickable`.

https://github.com/WordPress/gutenberg/blob/507d60b0db8dc4c056e3699a567394025d352d98/packages/dataviews/src/dataviews-layouts/hooks/use-clickable-item-props.ts#L25-L32

https://github.com/WordPress/gutenberg/blob/507d60b0db8dc4c056e3699a567394025d352d98/packages/dataviews/src/dataviews-layouts/hooks/use-clickable-item-props.ts#L19-L23

I'm fine with replacing both with something like this if you prefer:

```ts
onKeyDown: ! isClickable
? undefined
: ( event: React.KeyboardEvent ) => {
		if ( event.key === 'Enter' || event.key === '' ) {
			onClickItem( item );
		}
  },

onClick: ! isClickable
? undefined
: () => {
		onClickItem( item );
},
```

Reply:
yeah I just thought avoiding the handlers on things that are not "button" would be best.

Reply:
Fixed with c9e418b5fddc40020c9c677c3b29ae6dda162747!